nservicebus/upgrades/perfcounters-3to4.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Upgrade PerformanceCounters Version 3 to 4
+summary: Instructions on how to upgrade PerformanceCounters Version 3 to 4.
+component: PerfCounters
+reviewed: 2019-07-19
+isUpgradeGuide: true
+upgradeGuideCoreVersions:
+ - 7
+---
+
+NServiceBus.Metrics.PerformanceCounters 4.0 supports .NET Core, and deprecates the NServiceBus.Metrics.PerformanceCounters.MsBuild package.
+
+## .NET Core support
+
+NServiceBus.Metrics.PerformanceCounters now supports .NET Framework and .NET Core, however because performance counters are Windows only, the package only works in .NET Core on Windows. Attempting to run it on other architectures will result in a `PlatformNotSupportedException`.
+
+## MsBuild package deprecated
+
+In NServiceBus.Metrics.PerformanceCounters 4.0, the [NServiceBus.Metrics.PerformanceCounters.MsBuild](https://www.nuget.org/packages/NServiceBus.Metrics.PerformanceCounters.MsBuild) NuGet package is deprecated, and the PowerShell script used to create performance counters has been merged into the [NServiceBus.Metrics.PerformanceCounters](https://www.nuget.org/packages/NServiceBus.Metrics.PerformanceCounters) package.
+
+Version 4.0.0 of NServiceBus.Metrics.PerformanceCounters.MsBuild will be the last release of this package. When included in a project, it will produce this error at compile time:
+
+> The NServiceBus.Metrics.PerformanceCounters.MsBuild package is deprecated and should be removed from the project. The Powershell script to create performance counter objects is now included in the NServiceBus.Metrics.PerformanceCounters package.
+
+In all projects where the NServiceBus.Metrics.PerformanceCounters.MsBuild package is in use, it can be removed when upgrading to NServiceBus.Metrics.PerformanceCounters 4.0, and the PowerShell script will continue to be generated in the build output.
```

Comment:
```suggestion
In all projects where the NServiceBus.Metrics.PerformanceCounters.MsBuild package is in use, it can be removed when upgrading to NServiceBus.Metrics.PerformanceCounters 4.0, and the PowerShell script will continue to be included in the build output.
```